Prolangle/Languages/TypeScript.cs

```diff
@@ -0,0 +1,31 @@
+using Prolangle.Languages.Framework;
+
+namespace Prolangle.Languages;
+
+public class TypeScript : BaseLanguage
+{
+	private TypeScript()
+	{
+	}
+
+	public static TypeScript Instance { get; } = new();
+
+	public override Guid Id { get; } = Guid.NewGuid();
+	public override string Name => "TypeScript";
+	public override TypeSystem Typing => TypeSystem.Duck | TypeSystem.Structural | TypeSystem.Strong | TypeSystem.Static;
+	public override bool Compiled => true;
+	public override MemoryManagement MemoryManagement => MemoryManagement.TracingGarbageCollection;
+	public override SyntaxStyle SyntaxStyle => SyntaxStyle.C;
+
+	public override Applications KnownForBuilding =>
+		Applications.Client | Applications.Configuration |
+		Applications.Server | Applications.Web |
+		Applications.Scripts;
+
+	public override Paradigms Paradigms =>
+		Paradigms.EventDriven | Paradigms.Functional | Paradigms.Imperative |
+		Paradigms.Procedural | Paradigms.ObjectOriented | Paradigms.Generic;
```

Comment:
I vaguely recall that we didn't want to call a language `Procedural` and/or `Imperative` if it's also `ObjectOriented` (because otherwise, aren't _all_ OOP languages P and I?), but unsure